internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -670,26 +670,28 @@ func (b *BeatsMonitor) getHttpStreams(
 		endpoints := []interface{}{prefixedEndpoint(utils.SocketURLWithFallback(compInfo.ID, paths.TempDir()))}
 		name := sanitizeName(binaryName)
 
-		httpStream := map[string]interface{}{
-			idKey: fmt.Sprintf("%s-%s-1", monitoringMetricsUnitID, name),
-			"data_stream": map[string]interface{}{
-				"type":      "metrics",
-				"dataset":   dataset,
-				"namespace": monitoringNamespace,
-			},
-			"metricsets": []interface{}{"json"},
-			"hosts":      endpoints,
-			"path":       "/stats",
-			"namespace":  "agent",
-			"period":     metricsCollectionIntervalString,
-			"index":      indexName,
-			"processors": processorsForHttpStream(binaryName, compInfo.ID, dataset, b.agentInfo, compInfo.RuntimeManager),
-		}
-		if failureThreshold != nil {
-			httpStream[failureThresholdKey] = *failureThreshold
+		// Do not create http streams if runtime-manager is otel and binary is of beat type
+		if compInfo.RuntimeManager != component.OtelRuntimeManager || !strings.HasSuffix(binaryName, "beat") {
```

Comment:
If you switch the runtime while agent is running, does this work correctly? What I would expect is:
1. Going from standard to otel runtime, the http/metrics component is dropped.
2. Going from otel to standard runtime, the http/metrics component is restored.

Reply:
Why is the `"beat"` suffix check necessary? Shouldn't this entirely be covered by the first part of the if condition?

The beat suffix check would cover apm-server which uses libbeat but is not a beat, but it would't cover cloudbeat which will take some time to migrate into the collector as it isn't in the beats repository.

Reply:
> Because in case of apm-server - we send "all" apm-server related metrics to output.

Right, but we never plan to run `apm-server` as a receiver. I'm not sure that's even possible, why isn't the `compInfo.RuntimeManager != component.OtelRuntimeManager` check enough to handle this? Maybe I'm misunderstanding what this does, because I read it as "this component is currently configured to run the collector" which will never be true for `apm-server` AFAIK.

Reply:
This works as is, it's more of a nit that this check could be redundant